In [2]:
import pandas as pd

data = [
    ["DistilBERT (CPU)",           0.8113, 0.8021, 0.8088, 0.7991, 0.9678, 876.43],
    ["FastText CPU (preentrenado)",0.6078, 0.6154, 0.6022, 0.5983, 0.9308, 4.12],
    ["FastText GPU (preentrenado)",0.5931, 0.6202, 0.5902, 0.5971, 0.9295, 11.04],
    ["TF-IDF + XGBoost",           0.4510, 0.4947, 0.4470, 0.4106, 0.8521, 20.55],
    ["CNN-1D (Embeddings)",        0.3676, 0.3681, 0.3639, 0.3604, 0.8379, 20.39],
    ["BiLSTM (Word2Vec)",          0.3652, 0.3451, 0.3612, 0.3469, 0.8190, 374.85],
]

df_global = pd.DataFrame(data, columns=[
    "Modelo", "Accuracy", "Precision (Macro)", "Recall (Macro)",
    "F1-Score (Macro)", "ROC-AUC (OvR)", "Train Time (s)"
])

display(df_global)

# 💾 Guardar CSV
df_global.to_csv("metrics_global.csv", index=False, encoding='utf-8-sig')
print("✅ Resultados globales guardados en: metrics_global.csv")


Modelo  Accuracy  Precision (Macro)  Recall (Macro)  \
0             DistilBERT (CPU)    0.8113             0.8021          0.8088   
1  FastText CPU (preentrenado)    0.6078             0.6154          0.6022   
2  FastText GPU (preentrenado)    0.5931             0.6202          0.5902   
3             TF-IDF + XGBoost    0.4510             0.4947          0.4470   
4          CNN-1D (Embeddings)    0.3676             0.3681          0.3639   
5            BiLSTM (Word2Vec)    0.3652             0.3451          0.3612   

   F1-Score (Macro)  ROC-AUC (OvR)  Train Time (s)  
0            0.7991         0.9678          876.43  
1            0.5983         0.9308            4.12  
2            0.5971         0.9295           11.04  
3            0.4106         0.8521           20.55  
4            0.3604         0.8379           20.39  
5            0.3469         0.8190          374.85

✅ Resultados globales guardados en: metrics_global.csv


> ### Interpretación general
> - **DistilBERT** obtuvo el mejor desempeño global (F1 = 0.80, ROC-AUC = 0.97), mostrando una excelente capacidad de comprensión contextual.  
> - **FastText (CPU/GPU)** alcanzó un rendimiento intermedio (F1 ≈ 0.60) con tiempos de entrenamiento muy bajos, ideal para entornos de baja latencia.  
> - **TF-IDF + XGBoost** fue una buena baseline rápida pero con menor capacidad semántica.  
> - **CNN-1D** y **BiLSTM** mostraron underfitting, con menor F1 y dificultad para capturar dependencias largas.  
>
> **Conclusión:**  
> DistilBERT es el modelo más preciso y robusto,  
> mientras que FastText destaca por su eficiencia y velocidad de entrenamiento.
